### ISSUE TITLE
分享一个jieba加速make_vocab的方式，速度提升很多

### ISSUE THREAD

**Issue by 81549361** [Morizeyao/GPT2-Chinese#145]
from __future__ import unicode_literals
import argparse
import jieba
import jieba.posseg
import jieba.analyse
import thulac
import json

from tqdm import tqdm
from keras.preprocessing.text import Tokenizer

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('--raw_data_path', default='../data/train.json', type=str, required=False, help='原始训练语料')
    parser.add_argument('--vocab_file', default='vocab_processed.txt', type=str, required=False, help='生成vocab链接')
    parser.add_argument('--vocab_size', default=50000, type=int, required=False, help='词表大小')
    args = parser.parse_args()

    #lac = thulac.thulac(seg_only=True)
    #jieba.enable_parallel(8)
    tokenizer = Tokenizer(num_words=args.vocab_size)
    print('args:\n' + args.__repr__())
    print('This script is extremely slow especially for large corpus. Take a break.')

    f = open(args.raw_data_path, 'r')
    lines = json.load(f)
    for i, line in enumerate(tqdm(lines)):
        lines[i] = " ".join(jieba.cut(line, cut_all=True))

    tokenizer.fit_on_texts(lines)
    vocab = list(tokenizer.index_word.values())
    pre = ['[SEP]', '[CLS]', '[MASK]', '[PAD]', '[UNK]']
    vocab = pre + vocab
    with open(args.vocab_file, 'w') as f:
        for word in vocab[:args.vocab_size + 5]:
            f.write(word + '\n')

if __name__ == "__main__":
    main()

**Comment by walkingonthestreet** [comment]
用c++的jieba，我的数据集27G，用py的，得花2个月时间，c++的，几小时就分词完成了。

**Comment by Morizeyao** [comment]
感谢分享！
